Add README and other help text related to the commands (#11)
OnePunMan/thedasbot: README.md

# The "Do All Stuff" (DAS) Telegram Bot #

> The DAS bot is a webhook based telegram bot written in Python 3.7 deployed on AWS Lambda. It is capable of fetching jokes, fun facts, as well as as multiple choice trivia questions and fetch insights from Google Trends (As along as those APIs' remain functional).

Reviewer: "as" duplicated

> The DAS bot is a webhook based telegram bot written in Python 3.7 deployed on AWS Lambda. It is capable of fetching jokes, fun facts, as well as as multiple choice trivia questions and fetch insights from Google Trends (As along as those APIs' remain functional).

Reviewer 2: ```suggestion
The DAS bot is a webhook-based Telegram bot that is written in Python 3.7 and deployed on AWS Lambda. It is capable of telling jokes, sharing fun facts, hosting trivia sessions, and fetching insights from Google Trends.
```

## Usage ##
Interact with the bot **[@thedasbot](https://t.me/thedasbot)** on Telegram. It supports both private and group chats.

## Commands ##
### Help ###
Pulls up the list of available commands.

**Syntax:** `/help`

### Trivia ###
Gets a random, interactable multiple choice question from [Open Trivia Database](https://opentdb.com/). 

**Syntax:** `/trivia [ <category> | any | help ] [ <difficulty> | any ]`

> - `/trivia help` brings up the list of available categories and difficulties. Categories do not need to exactly match what's listed, as long it's enough to uniquely identify it. Leaving any arguments blank will default to `any`, which selects a random category/difficulty.

Reviewer 2: ```suggestion
- `/trivia help` brings up the list of available categories and difficulties. Categories support fuzzy matching. All arguments default to `any`, which selects a random category/difficulty.
```

**Examples**

* `/trivia` - Question from random category and difficulty

* `/trivia history` - Question from the "History" category at random difficulty

* `/trivia anime easy` - Easy question from the "Entertainment: Japanese Anime & Manga" category

> * `/trivia any hard` - Hard question from a random category

Reviewer 3: Can you include pictures of the bot from Telegram here? It would be nice to see the UI here on Github,

Reviewer 2: I personally wouldn't add UI screenshots, it's not really the norm unless you actually have custom UI (which you don't, you're using Telegram)

### Trends ###

> Gets insights from Google Trends; capble graphing trends, pulling top trends, and realtime trending searches. It uses the [pytrends](https://github.com/GeneralMills/pytrends) library.

Reviewer: use "real-time"

> Gets insights from Google Trends; capble graphing trends, pulling top trends, and realtime trending searches. It uses the [pytrends](https://github.com/GeneralMills/pytrends) library.

Reviewer 2: ```suggestion
Fetches insights from Google Trends. Features the ability to graph trends by keyword and list top / trending keywords by year. All communication is handled using [pytrends](https://github.com/GeneralMills/pytrends).
```

    * `trending` - Gets the current trending searches for a specific country. (Default US)

**Examples**
* `/trends graph BMO "Bank of America" RBC` - Graphs the search trends of "BMO", "Bank of America" and "RBC" on the same plot for the past year.
* `/trends graph Honda Toyota --time 6w` - Graphs the search trends of "Honda" and "Toyota" on the same plot in the past 6 weeks.
* `/trends graph GitHub -t 4m` - Graphs the search trends of "GitHub" in the past 4 months.
* `/trends top` - If we're in year 2020, it gets the top searches for year 2019
* `/trends top 2012` - Gets the top searches for 2012
* `/trends trending` - Gets the realtime trending searches in the USA (default region)
* `/trends trending canada` - Gets the realtime trending searches in the Canada
* `/trends trending "new zealand"` - Gets the realtime trending searches in New Zealand

### Joke ###
Fetches a random joke from [icanhazdadjoke](https://icanhazdadjoke.com/).

**Syntax:** `/joke`

### Fact ###
Fetches a random fun fact from [Mental Floss](https://www.mentalfloss.comamazingfactgenerator).

**Syntax:** `/fact`
    
## Contributing ##

> Got an feature idea? Make a pull request! New features and improvements are always welcome! New commands should inherit the `BotCommand` class, override at least the `execute` method, and placed inside the `commands/` directory. Also, the command must be added to the `AVAILABLE_COMMANDS` dictionary in `command_parser.py` to be recognized by the parser.

Reviewer 2: ```suggestion
New features and improvements are always welcome! Commands should inherit the `BotCommand` class, override the `execute` method, and be placed inside the `commands/` directory. Also, the command must be added to the `AVAILABLE_COMMANDS` constant in `command_parser.py` to be recognized by the parser.
```

### Local Testing ###

> My local webhook setup uses [bottle](https://pypi.org/project/bottle/) and [ngrok](https://ngrok.com/download). This [guide](https://hackernoon.com/serverless-telegram-bot-on-aws-lambda-851204d4236c) is what I used to setup my local environment. If you want to use the same setup, run `install_local`, then `python3 local.py` after you have ngrok running and webhook set.

Reviewer 2: ```suggestion
To test the project locally:
1. Setup [bottle](https://pypi.org/project/bottle/) and [ngrok](https://ngrok.com/download)
2. Setup the local environment using the following [guide](https://hackernoon.com/serverless-telegram-bot-on-aws-lambda-851204d4236c)
3. Launch `ngrok` and update the webhook to the project
4. Run `install_local`, then `python3 local.py`
```
